internal/checker/symboltracker.go

```diff
@@ -0,0 +1,140 @@
+package checker
+
+import (
+	"github.com/microsoft/typescript-go/internal/ast"
+	"github.com/microsoft/typescript-go/internal/modulespecifiers"
+	"github.com/microsoft/typescript-go/internal/nodebuilder"
+)
+
+type SymbolTrackerImpl struct {
+	context            *NodeBuilderContext
+	inner              nodebuilder.SymbolTracker
+	DisableTrackSymbol bool
+	tchost             Host
+}
+
+func NewSymbolTrackerImpl(context *NodeBuilderContext, tracker nodebuilder.SymbolTracker, tchost Host) *SymbolTrackerImpl {
+	var inner nodebuilder.SymbolTracker
+	if tracker != nil {
+		inner = tracker.GetInnerSymbolTracker()
+		if inner == nil {
+			inner = tracker
+		}
+	}
+
+	return &SymbolTrackerImpl{context, inner, false, tchost}
+}
+
+func (this *SymbolTrackerImpl) GetModuleSpecifierGenerationHost() modulespecifiers.ModuleSpecifierGenerationHost {
```

Comment:
`this` 🤨

Reply:
I guess this is the _only_ class in the old codebase, but I still don't like it 😄